src/modules/launcher/Plugins/Microsoft.Plugin.Registry/Main.cs

```diff
@@ -0,0 +1,171 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using ManagedCommon;
+using Microsoft.Plugin.Registry.Classes;
+using Microsoft.Plugin.Registry.Constants;
+using Microsoft.Plugin.Registry.Helper;
+using Microsoft.Plugin.Registry.Properties;
+using Wox.Plugin;
+
+[assembly: InternalsVisibleTo("Microsoft.Plugin.Registry.UnitTest")]
+
+namespace Microsoft.Plugin.Registry
+{
+    /// <summary>
+    /// Main class of this plugin that implement all used interfaces
+    /// </summary>
+    public class Main : IPlugin, IContextMenu, IPluginI18n, IDisposable
+    {
+        /// <summary>
+        /// The name of this assembly
+        /// </summary>
+        private readonly string _assemblyName;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data)
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _defaultIconPath;
+
+        /// <summary>
+        /// Indicate that the plugin is disposed
+        /// </summary>
+        private bool _disposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            _assemblyName = Assembly.GetExecutingAssembly().GetName().Name ?? GetTranslatedPluginTitle();
+            _defaultIconPath = "Images/reg.light.png";
+        }
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            // Any base registry key have more than two characters
+            if (query?.Search is null || query.Search.Length < 3)
+            {
+                return new List<Result>(0);
+            }
+
+            var searchForValueName = QueryHelper.GetQueryParts(query.RawQuery, out var queryKey, out var queryValueName);
+
+            var (baseKeyList, subKey) = RegistryHelper.GetRegistryBaseKey(queryKey);
+            if (baseKeyList is null)
+            {
+                // no base key found -> test is the key starts with "HKE" or "HKEY" (case-insensitive)
```

Comment:
If `HKE` is normally a misspelling, I'd suggest adding:
```
"HKE" or "HKEY"
```
to `patterns.txt` instead of adding `HKE` to `expect.txt`